### PR TITLE
New test: GPG signature of the OSTree

### PR DESCRIPTION
**Pull request by ljozsa**
Hi folks,

please review the pull request.

Thanks,
Ladislav

### CODE REVIEW

File: steps/redhat.py [hunk 1]
```diff
@@ -166,3 +168,57 @@ def step_impl(context):
...
+
+@when(u"import Red Hat's release key 2 to the superuser's keyring succeeds")
+def step_impl(context):
+    assert context.remote_cmd("command",
+           module_args="gpg --import /etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release"), "import of Red Hat's release key 2 failed"
+
+@then(u"verify if Red Hat's release key 2 matches public fingerprint")
+def step_impl(context):
+	gpg_out = context.remote_cmd("command", module_args="gpg --fingerprint")[0]['stdout']
+	m = re.search('567E 347A D004 4ADE 55BA  8A5F 199E 2F91 FD43 1D51',gpg_out)
```
Comment (miabbott) [inline steps/redhat.py]: Perhaps create a variable for the public fingerprint so that it is more easily reused?

File: steps/redhat.py [hunk 2]
```diff
@@ -166,3 +168,57 @@ def step_impl(context):
...
+	assert m.group(0) == '567E 347A D004 4ADE 55BA  8A5F 199E 2F91 FD43 1D51', "the imported Red Hat's release key 2 does not match its public fingerprint"
+
+@when(u'download "{file}" script with sha256sum "{sha256sum}" finishes')
+def step_impl(context, file, sha256sum):
+	download_status = context.remote_cmd("get_url", module_args="url=" + file + " dest=/home/cloud-user/gpgverify.py force=yes sha256sum=" + sha256sum)
```
Comment (miabbott) [inline steps/redhat.py]: Hardcoding `dest=/home/cloud-user/gpgverify.py` could cause problems if this code is ever reused.

Saving the script to `/tmp` should work.

File: steps/redhat.py [hunk 3]
```diff
@@ -166,3 +168,57 @@ def step_impl(context):
...
+	if not LooseVersion(running_version) < LooseVersion(version): 
+		context.scenario.skip(reason='OSTree version is greater or equal than ' + version + '. Skipping this scenario')
+
+@then(u'use the gpgverify.py script to verify gpg signatures')
+def step_impl(context):
+	atomic_host_status =  context.remote_cmd(cmd='shell',
+                    module_args='atomic host status | grep "^*"')[0]['stdout']
+	treeid = " ".join(atomic_host_status.split()).split()[4]
+	chmod_status =  context.remote_cmd(cmd='file',
+                    module_args='path=/home/cloud-user/gpgverify.py mode=0755')
```
Comment (miabbott) [inline steps/redhat.py]: See my previous comment about hardcoding to `/home/cloud-user`

File: steps/redhat.py [hunk 4]
```diff
@@ -166,3 +168,57 @@ def step_impl(context):
...
+	assert chmod_status[0].has_key('failed') == False, "attempt to chmod 0755 /home/cloud-user/gpgverify.py failed"
+	gpgverify_out =  context.remote_cmd(cmd='command',
+                    module_args='/home/cloud-user/gpgverify.py /sysroot/ostree/repo ' + treeid)
```
Comment (miabbott) [inline steps/redhat.py]: See my previous comment about hardcoding to `/home/cloud-user`

File: steps/redhat.py [hunk 5]
```diff
@@ -166,3 +168,57 @@ def step_impl(context):
...
+	assert gpgverify_out, "the gpgverify.py script crashed, the OSTree isn't signed"
+	m = re.search('(?<==> )(\w+)',gpgverify_out[0]['stdout'])
+	commit_path = '/sysroot/ostree/repo/objects/' + m.group(0)[:2] + '/' + m.group(0)[2:] + '.commit'
+	gpg_out =  context.remote_cmd(cmd='command',
+                    module_args='gpg --verify sig.0 ' + commit_path)
+	assert gpg_out, "verification of the gpg signature has failed"
+	m = re.search('(?<=Primary key fingerprint: )(.*)',gpg_out[0]['stderr'])
+	primary_key = m.group(0)
+	assert primary_key == '567E 347A D004 4ADE 55BA  8A5F 199E 2F91 FD43 1D51', "the OSTree signature does not match Red Hat's release key 2"
```
Comment (miabbott) [inline steps/redhat.py]: Consider making the key value a variable

File: steps/redhat.py [hunk 6]
```diff
@@ -166,3 +168,57 @@ def step_impl(context):
...
+
+@then(u'use ostree show command to verify gpg signatures')
+def step_impl(context):
+	atomic_host_status =  context.remote_cmd(cmd='shell',
+                    module_args='atomic host status | grep "^*"')[0]['stdout']
+	tree_version = " ".join(atomic_host_status.split()).split()[3]
+	treeid = " ".join(atomic_host_status.split()).split()[4]
+	signature_status =  context.remote_cmd(cmd='shell',
+                    module_args='ostree show ' + treeid + ' | grep 199E2F91FD431D51')
```
Comment (miabbott) [inline steps/redhat.py]: Consider making this key value a variable, as well

**Comment by miabbott** [comment]
Overall, this test looks good.  Just a couple of places I see that could be improved.

**Comment by ljozsa** [comment]
@miabbott I have fixed the issues you have found. Please review again.

**Comment by miabbott** [comment]
:+1:

**Pull request merged**